generic3g/tests/Test_VerticalLinearMap.pf

```diff
@@ -0,0 +1,47 @@
+#include "MAPL_TestErr.h"
+
+module Test_VerticalLinearMap
+
+   use mapl3g_CSR_SparseMatrix, only: SparseMatrix_sp => CSR_SparseMatrix_sp
+   use mapl3g_VerticalLinearMap, only: compute_linear_map_fixedlevels_to_fixedlevels
+   use mapl3g_VerticalLinearMap, only: apply_linear_map
+   use funit
+   use, intrinsic :: iso_fortran_env, only: REAL32
+
+   implicit none
+
+contains
+
+   @test
+   subroutine test_linear_map_fixedlevels_to_fixedlevels()
+
+      real(REAL32), allocatable :: vcoord_src(:), vcoord_dst(:)
+      real(REAL32), allocatable :: fin(:), fout(:)
+      ! real(REAL32), allocatable :: matrix(:, :)
+      type(SparseMatrix_sp) :: matrix
+      integer :: status
+
+      vcoord_src = [30., 20., 10.]
+      vcoord_dst = [20., 10.]
+      call compute_linear_map_fixedlevels_to_fixedlevels(vcoord_src, vcoord_dst, matrix, _RC)
+      fin = [7., 8., 3.]
+      call apply_linear_map(matrix, fin, fout)
+      @assertEqual(fout, [8., 3.])
+
+      vcoord_src = [30., 20., 10.]
+      vcoord_dst = [25., 15.]
+      call compute_linear_map_fixedlevels_to_fixedlevels(vcoord_src, vcoord_dst, matrix, _RC)
+      fin = [7., 8., 3.]
+      call apply_linear_map(matrix, fin, fout)
+      @assertEqual(fout, [7.5, 5.5])
+
+      vcoord_src = [30., 20., 10.]
+      vcoord_dst = [28., 11.]
+      call compute_linear_map_fixedlevels_to_fixedlevels(vcoord_src, vcoord_dst, matrix, _RC)
+      fin = [7., 8., 3.]
+      call apply_linear_map(matrix, fin, fout)
+      @assert_that(norm2(fout - [7.2, 3.5]) < 2.4e-7, is(true()))
```

Comment:
There is an option here for 
```
@assert_that(f_expected, is(nearly_equal_to(..., tol)))
```
Been a while since I used it though, so am not positive that the array version is implemented.

Reply:
Certainly you want easier numbers to compute in your head though here.  In the previous test it was debatable.  But here, definitely make the expected output easy to compute in your head.